src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -115,46 +111,32 @@ internal void AssertBalanced()
 #endif
         }
 
-        private class Node
+        private abstract class Node
         {
-            public readonly K key;
+            public readonly K Key;
             public V Value;
 
-            public Node(K key, V value)
+            protected Node(K key, V value)
             {
-                this.key = key;
+                this.Key = key;
                 this.Value = value;
             }
 
-            public virtual Node Next
-            {
-                get
-                {
-                    return null;
-                }
-            }
+            public virtual Node Next => null;
         }
 
-        private class NodeLinked : Node
+        private sealed class NodeLinked : Node
         {
-            public readonly Node next;
-
             public NodeLinked(K key, V value, Node next)
                 : base(key, value)
             {
-                this.next = next;
+                this.Next = next;
             }
 
-            public override Node Next
-            {
-                get
-                {
-                    return next;
-                }
-            }
+            public override Node Next { get; }
         }
 
-        private class AvlNodeHead : AvlNode
+        private sealed class AvlNodeHead : AvlNode
         {
             public Node next;
```

Comment:
:question: Our coding guidelines recommend that this field name is capitalized, but that would introduce a conflict with the property `Next` below. What would you recommend?